src/App.js

```diff
@@ -1,25 +1,82 @@
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+import React from 'react';
+import axios from 'axios';
+import Container from 'react-bootstrap/Container';
+import Image from 'react-bootstrap/Image';
+import Button from 'react-bootstrap/Button'
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchQuery: '',
+      location: '',
+      latitude: '',
+      longitude: '',
+      error: false,
+    }
+  }
+
+  getLocation = async () => {
+    const locationURL = `https://us1.locationiq.com/v1/search.php?key=${process.env.REACT_APP_CITY_KEY}&q=${this.state.searchQuery}&format=json`;
+
+    try {
+      const response = await axios.get(locationURL);
+      const location = response.data[0].display_name;
+      const latitude = response.data[0].lat;
+      const longitude = response.data[0].lon;
+
+      console.log(latitude, longitude, 'CONSOLE LOG')
+
+      this.setState({
+        location, // or location:location
+        latitude,
+        longitude,
+        error: false,
+      });
+
+      // this.showMap();
+
+    } catch (error) {
+
+      console.error('Unable to find city', 
+      this.state.searchQuery);
+
+      this.setState({ error: true });
+    }
+  }
+
+  render() {
+
+    return (
+      <>
+        <input onChange={(event) => this.setState({ searchQuery: event.target.value })} placeholder="search for a city"></input>
+        <Button onClick={this.getLocation} variant="warning">Explore!</Button>
```

Comment:
Hello there, nice work so far on this! 👍 

Feedback: Remove the setState call out of render. This is not recommended as it can cause some strange things to happen (infinite loops, can make component less maintainable etc. I encourage you to read more about this when ya have time :)

Instead, create a method outside of render to handle event and then pass the handler to onChange